instrumentation/kafka-clients-0.11/src/main/java/io/opentelemetry/instrumentation/auto/kafkaclients/KafkaConsumerTracer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.auto.kafkaclients;
+
+import static io.opentelemetry.instrumentation.api.decorator.BaseDecorator.extract;
+import static io.opentelemetry.instrumentation.auto.kafkaclients.TextMapExtractAdapter.GETTER;
+import static io.opentelemetry.trace.Span.Kind.CONSUMER;
+
+import io.grpc.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.attributes.SemanticAttributes;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+
+public class KafkaConsumerTracer extends BaseTracer {
+  public static final KafkaConsumerTracer TRACER = new KafkaConsumerTracer();
+
+  public Span startSpan(ConsumerRecord<?, ?> record) {
+    long now = System.currentTimeMillis();
+
+    Span span =
+        tracer
+            .spanBuilder(spanNameOnConsume(record))
+            .setSpanKind(CONSUMER)
+            .setParent(extractParent(record))
+            .setStartTimestamp(TimeUnit.MILLISECONDS.toNanos(now))
```

Comment:
Why do we need to specify start timestamp to current time ourselves here?

Reply:
It is used later in `onConsume` method to calculate `record.queue_time_ms`